interface/forms/LBF/new.php

```diff
@@ -1531,6 +1589,94 @@ function warehouse_changed(sel) {
                 echo "</div>\n";
             } // End Diagnoses Section
 
+            if ($LBF_REFERRALS_SECTION) {
+                // Create the checkbox and div for the Referrals Section.
+                echo "<br /><span class='bold'><input type='checkbox' name='form_cb_referrals' value='1' " .
+                    "onclick='return divclick(this, \"div_referrals\");'";
+                if ($display_style == 'block') {
+                    echo " checked";
+                }
+                echo " /><b>" . xlt('Referrals') . "</b></span>\n";
+                echo "<div id='div_referrals' class='section' style='display:" . attr($display_style) . ";'>\n";
+                echo "<center>\n";
+                // $display_style = 'none';
+
+                // Generate a table row for each referral in the visit.
+                echo "<table cellpadding='0' cellspacing='5' id='referrals_table'>\n";
+                echo " <tr>\n";
+                echo "  <td class='bold'>" . xlt('Date') . "&nbsp;</td>\n";
+                echo "  <td class='bold'>" . xlt('Type') . "&nbsp;</td>\n";
+                echo "  <td class='bold'>" . xlt('Reason') . "&nbsp;</td>\n";
+                echo "  <td class='bold'>" . xlt('Referred To') . "&nbsp;</td>\n";
+                echo "  <td class='bold'>" . xlt('Requested Service') . "</td>\n";
+                echo " </tr>\n";
+
+                $refres = sqlStatement(
+                    "SELECT f.form_id, " .
+                    "d1.field_value AS refer_external, " .
+                    "d2.field_value AS body, " .
+                    "lo.title AS refer_type, " .
+                    "ut.organization, " .
+                    "CONCAT(ut.fname,' ', ut.lname) AS referto_name, " .
+                    "d4.field_value AS refer_related_code, " .
+                    "d5.field_value AS refer_date " .
+                    "FROM forms AS f " .
+                    "LEFT JOIN lbf_data AS d1 ON d1.form_id = f.form_id AND d1.field_id = 'refer_external' " .
+                    "LEFT JOIN list_options AS lo ON list_id = 'reftype' and option_id = d1.field_value " .
+                    "LEFT JOIN lbf_data AS d2 ON d2.form_id = f.form_id AND d2.field_id = 'body' " .
+                    "LEFT JOIN lbf_data AS d3 ON d3.form_id = f.form_id AND d3.field_id = 'refer_to' " .
+                    "LEFT JOIN users AS ut ON ut.id = d3.field_value " .
+                    "LEFT JOIN lbf_data AS d4 ON d4.form_id = f.form_id AND d4.field_id = 'refer_related_code' " .
+                    "LEFT JOIN lbf_data AS d5 ON d5.form_id = f.form_id AND d5.field_id = 'refer_date' " .
+                    "WHERE " .
+                    "f.pid = ? AND f.encounter = ? AND f.formdir = 'LBFref' AND f.deleted = 0 " .
+                    "ORDER BY refer_date, f.form_id",
+                    array($pid, $encounter)
```

Comment:
dang, now that is a sql query :)

![2DV (1)](https://user-images.githubusercontent.com/278968/112104332-4bd45b80-8b68-11eb-9021-eee0defa668d.gif)